datafusion/core/src/execution/context/mod.rs

```diff
@@ -934,7 +934,26 @@ impl SessionContext {
             .build()?,
         ))
     }
-
+    /// Create a [`DataFrame`] for reading a [`Vec[`RecordBatch`]`]
+    pub fn read_batches(
+        &self,
+        batches: impl IntoIterator<Item = RecordBatch>,
+    ) -> Result<DataFrame> {
+        // check schema uniqueness
+        let mut batches = batches.into_iter().peekable();
+        let schema: SchemaRef = batches.peek().unwrap().schema().clone();
```

Comment:
Is there any way we can avoid an unwrap here?

Reply:
Perhaps something like

```suggestion
        let schema: SchemaRef = if let Some(schema) = batches.peek() {
          schema.clone()
        } else {
          // use empty schema)
          Arc::new(Schema::new())
        };
```

Reply:
Got it!